### ISSUE TITLE
wrong code

### ISSUE THREAD

**Issue by Guanyu-Zhou** [qyp2000/VARSR#19]
sos = torch.cat((sos, cond_BD.unsqueeze(1)), dim=1)
sos = sos.expand(B, self.first_l, -1) + self.pos_start.expand(B, self.first_l, -1)

maybe something wrong

**Comment by Guanyu-Zhou** [comment]
It seems there are many minor errors in the code. Could it be because of my poor coding skills? Why does it become 2 after concatenation on dimension 1? And can the broadcasting mechanism still be utilized?

**Comment by quyp2000** [comment]
Hello, please provide more detailed information to help us locate your issue. This line of code looks correct.

**Comment by Guanyu-Zhou** [comment]
> 您好，请提供更详细的信息，以帮助我们找到您的问题。这行代码看起来是正确的。

Sorry, I failed to complete the comparison with the original VAR. I overlooked your changes to self.first_l. Therefore, I encountered an error when modifying my own code. This is not your fault.
